jax/_src/numpy/reductions.py

```diff
@@ -2482,12 +2492,80 @@ def _quantile(a: Array, q: Array, axis: int | tuple[int, ...] | None,
   else:
     axis = canonicalize_axis(axis, a.ndim)
 
+  q, = promote_dtypes_inexact(q)
+
   q_shape = q.shape
   q_ndim = q.ndim
   if q_ndim > 1:
     raise ValueError(f"q must be have rank <= 1, got shape {q.shape}")
 
   a_shape = a.shape
+  # Handle weights
+  if weights is None:
+    a, = promote_dtypes_inexact(a)
+  else:
+    a, weights = promote_dtypes_inexact(a, weights)
+    a_shape = a.shape
+    w_shape = np.shape(weights)
+    if w_shape != a_shape:
+      if len(w_shape) != 1:
+        raise ValueError("1D weights expected when shapes of a and weights differ.")
+      if axis is None:
+        raise TypeError("Axis must be specified when shapes of a and weights differ.")
+      if w_shape[0] != a_shape[axis]:
+        raise ValueError("Length of weights not compatible with specified axis.")
+      resh = [1] * a.ndim
+      resh[axis] = w_shape[0]
+      weights = lax.expand_dims(weights, axis)
+    weights = _broadcast_to(weights, a.shape)
```

Comment:
Should this be in the `if` block? If `weights.shape` is already equal to `a.shape`, this is not needed.